### PR TITLE
[ACIX-254] Send metrics on failure alerts

### PR DESCRIPTION
**Pull request by CelianR**
### What does this PR do?

Send metrics when failing job alerts are sent. This will be modified when job will be filtered by team

Part of [this RFC](https://docs.google.com/document/d/1gdhZJ_eDmdFixL1AEYMWMXX7vcuYRvvQQ8ketHi7Gqw/edit?pli=1)

### CODE REVIEW

File: tasks/notify.py [hunk 1]
```diff
@@ -436,11 +436,39 @@ def update_statistics(job_executions: PipelineRuns):
...
 
     if message:
         send_slack_message("#agent-platform-ops", message)
+        print('Sent message')
+
+        # TODO : When these notifications will be sent to many teams, update the team tag (https://datadoghq.atlassian.net/browse/ACIX-255)
+        owner = '@datadog/agent-all'
+        team_name = owner.removeprefix('@datadog/')
+
+        # Create metrics for consecutive and cumulative alerts
```
Comment (chouetz) [inline tasks/notify.py:456]: What will be the objective of this metric? Will we put it in a dashboard or a monitor?

**Review by chouetz — APPROVED** [review]

File: tasks/notify.py (hunk 1 above)
Comment (CelianR) [reply tasks/notify.py:456]: It is to monitor how many notifications are sent to the teams, I've added a metric with the current notification system. We will be able to compare them

File: tasks/notify.py [hunk 2]
```diff
@@ -436,11 +436,39 @@ def update_statistics(job_executions: PipelineRuns):
...
+                unit="notification",
+                value=len(alert_jobs[alert_type].failures),
+            )
+            for alert_type in ("consecutive", "cumulative")
+            if len(alert_jobs[alert_type].failures) > 0
+        ]
+
+    if metrics:
+        send_metrics(metrics)
+        print('Sent metrics')
```
Comment (FlorentClarret) [inline tasks/notify.py:471]:
```suggestion
        print('Metrics sent')
```

File: tasks/notify.py [hunk 3]
```diff
@@ -436,11 +436,39 @@ def update_statistics(job_executions: PipelineRuns):
...
+        team_name = owner.removeprefix('@datadog/')
+
+        # Create metrics for consecutive and cumulative alerts
+        metrics += [
+            create_count(
+                metric_name=f"datadog.ci.failed_job_alerts.{alert_type}",
+                timestamp=timestamp,
+                tags=[f"team:{team_name}", "repository:datadog-agent"],
+                unit="notification",
+                value=len(alert_jobs[alert_type].failures),
```
Comment (FlorentClarret) [inline tasks/notify.py:463]: IIUC, in a next PR we'll loop over a list of teams to send this metric. How will we be able to get the right value for the right team from `alert_jobs[alert_type].failures`?

**Review by FlorentClarret — COMMENTED** [review]
Left some questions

File: tasks/notify.py (hunk 3 above)
Comment (CelianR) [reply tasks/notify.py:463]: See [this PR](https://github.com/DataDog/datadog-agent/pull/26326), I'll rebase this PR on top of it today

**Review by FlorentClarret — APPROVED** [review]

**Review by FlorentClarret — APPROVED** [review]

**Pull request merged**